actors/core/src/main/java/cloud/orbit/actors/runtime/DefaultClassDictionary.java

```diff
@@ -163,14 +170,18 @@ private void loadClassInfo(final String resourceName, final InputStream stream)
                 {
                     final ConcurrentHashMap<Integer, String> actualValue = new ConcurrentHashMap<>();
                     long start = System.currentTimeMillis();
-                    for (String candidate : new FastClasspathScanner().scan().getNamesOfAllClasses())
+                    try (ScanResult scanResult = new ClassGraph().enableInterClassDependencies().scan())
                     {
-                        String old = actualValue.put(candidate.hashCode(), candidate);
-                        if (old != null)
+                        for (String candidate : scanResult.getAllClasses().getNames())
                         {
-                            logger.info("Found more than one class with hashCode #" + candidate.hashCode() + " replacing " + old + " with " + candidate);
+                            String old = actualValue.put(candidate.hashCode(), candidate);
+                            if (old != null)
+                            {
+                                logger.info("Found more than one class with hashCode #" + candidate.hashCode() + " replacing " + old + " with " + candidate);
```

Comment:
(You're right, the mobile interface doesn't even have a Submit Review button...)

Reply:
The class name is already unique (within a classloader) and deterministic. Why use the hashCode? This is bound to fail at some point.

Reply:
The entire way ids are used is interesting to say the least..

```java
        String className = idToName.get(id);
        if (className == null)
        {
            // reading the class name saved in the file system by the annotation processor
            String fileName = "/META-INF/services/orbit/classId/"
                    + (Math.abs(classId) % 10)
                    + "/" + (Math.abs(classId / 10) % 10)
                    + ((classId < 0) ? "/m" : "/") + Math.abs(classId) + ".name";

            InputStream resourceAsStream = getClass().getResourceAsStream(fileName);
```

while I do agree it is fragile I believe it's out of scope to refactor in this pull request :stuck_out_tongue_winking_eye:

Reply:
I agree this is sketchy. 

We can't just use a counter even if scanning was deterministic because there is no guarantee the classpath is the same on each node (asymmetric node deployment is a supported scenario).

The good news is that this is resolved in Orbit 2.x as we use the FQN in all cases rather than doing any hashing etc. 
As sketchy as this is, it has been there for several years so I am not convinced it is worth the investment to resolve given Orbit 2.x is due in the next few months.